File: algorithms/fair-rations/fairRations.php
<?php
  function fairRations($B) {
    $n = 0;
    for ($i = 0; $i < count($B); $i++) {
      if ($B[$i] % 2 <> 0 && !($i == count($B) - 1))  {
        $B[$i]++;
        $B[$i + 1]++;
        $n += 2;
      }
    }
    
    $j = count($B) - 1;
    
    for ($i = 0; $i < count($B); $i++) {
      if ($i > $j) {
        break;
      }
      if ($B[$i] % 2 <> 0 || $B[$j] % 2 <> 0) {
        return "NO";
      }
            
      $j--;
    }

    return strval($n);
  }
?>

File: algorithms/fair-rations/fairRations.py
def fairRations(B):
  n = 0
  for i in range(len(B)):
    if (B[i] % 2 != 0) and (not (i == len(B) - 1)):
      B[i] += 1
      B[i + 1] += 1
      n += 2

  j = len(B) - 1
  
  for i in range(len(B)):
    if (B[i] % 2 != 0) or (B[j] % 2 != 0):
      return "NO"
    j -= 1
    
  return str(n)